src/lib/state.ts

```diff
@@ -113,6 +124,43 @@ export async function fetchBalances(
 	return { icp: icpBalanceE8s, nicp: nicpBalanceE8s, wtn: wtnBalanceE8s };
 }
 
+const identity = Ed25519KeyIdentity.generate();
+export const principal = identity.getPrincipal();
+
+const agent = Promise.resolve(new HttpAgent({ host: HOST, identity })).then(async (agent) => {
+	await agent.fetchRootKey();
+	agent.addTransform('query', makeNonceTransform());
+	return agent;
+});
+
+async function createSecp256k1IdentityActor(
+	canisterId: Principal,
+	idl: IDL.InterfaceFactory,
+	seed?: number
+): Promise<any> {
+	let seed1: Uint8Array | undefined;
+	if (seed) {
+		seed1 = new Uint8Array(new Array(32).fill(0));
+		seed1[0] = seed;
+	}
```

Comment:
```suggestion
seed1 = new Uint8Array(new Array(32).fill(0));
```

Reply:
Yeah I changed the code completely in the new preview